pkg/process/events/store_test.go

```diff
@@ -0,0 +1,284 @@
+
+	pushSync(t, s, e3)
+	require.Equal(t, 1, s.head)
+	require.Equal(t, 3, s.tail)
+	require.Equal(t, 2, s.size())
+
+	data, err = s.Pull(ctx, timeout)
+	assert.NoError(t, err)
+	require.Equal(t, []*model.ProcessEvent{e2, e3}, data)
+	require.Equal(t, 3, s.head)
+	require.Equal(t, 3, s.tail)
+	require.Equal(t, 0, s.size())
+}
+
+func TestRingStoreWithLoop(t *testing.T) {
+	now := time.Now()
+	ctx := context.Background()
+	timeout := time.Second
+
+	cfg := config.Mock()
+	cfg.Set("process_config.event_collection.store.max_items", 3)
+	store, err := NewRingStore(&statsd.NoOpClient{})
+	require.NoError(t, err)
+
+	s, ok := store.(*RingStore)
+	require.True(t, ok)
+
+	e1 := model.NewMockedProcessEvent(model.Exec, now, 23, "/usr/bin/curl", []string{"curl", "localhost:6062"})
+	e2 := model.NewMockedProcessEvent(model.Exit, now, 23, "/usr/bin/curl", []string{"curl", "localhost:6062"})
+	e3 := model.NewMockedProcessEvent(model.Exit, now, 23, "/usr/bin/ls", []string{"ls", "-lah"})
+
+	// Initialize store with len(buffer)-1 events that have already been consumed
+	s.head = 2
+	s.tail = 2
+	require.Equal(t, 0, s.size())
+
+	s.Run()
+	defer s.Stop()
+
+	// Push 1 elements to reach end of buffer - no data loss
+	pushSync(t, s, e1)
+	require.Equal(t, 2, s.head)
+	require.Equal(t, 0, s.tail)
+	require.Equal(t, 1, s.size())
+
+	// Push 2 more elements - buffer full - no data loss
+	pushSync(t, s, e2)
+	require.Equal(t, 2, s.head)
+	require.Equal(t, 1, s.tail)
+	require.Equal(t, 2, s.size())
+
+	pushSync(t, s, e3)
+	require.Equal(t, 2, s.head)
+	require.Equal(t, 2, s.tail)
+	require.Equal(t, 3, s.size())
+
+	// Retrieve all events from buffer
+	data, err := s.Pull(ctx, timeout)
+	assert.NoError(t, err)
+	require.Equal(t, []*model.ProcessEvent{e1, e2, e3}, data)
+	require.Equal(t, 2, s.head)
+	require.Equal(t, 2, s.tail)
+	require.Equal(t, 0, s.size())
+}
+
+func TestRingStoreWithDroppedData(t *testing.T) {
+	now := time.Now()
+	ctx := context.Background()
+	timeout := time.Second
+
+	cfg := config.Mock()
+	cfg.Set("process_config.event_collection.store.max_items", 3)
+	store, err := NewRingStore(&statsd.NoOpClient{})
+	require.NoError(t, err)
+
+	s, ok := store.(*RingStore)
+	require.True(t, ok)
+
+	s.Run()
+	defer s.Stop()
+
+	e1 := model.NewMockedProcessEvent(model.Exec, now, 23, "/usr/bin/curl", []string{"curl", "localhost:6062"})
+	e2 := model.NewMockedProcessEvent(model.Exit, now, 23, "/usr/bin/curl", []string{"curl", "localhost:6062"})
+	e3 := model.NewMockedProcessEvent(model.Exit, now, 23, "/usr/bin/ls", []string{"ls", "-lah"})
+	e4 := model.NewMockedProcessEvent(model.Exit, now, 23, "/usr/bin/ls", []string{"ls", "-lah"})
+
+	// Fill up buffer
+	pushSync(t, s, e1)
+	pushSync(t, s, e2)
+	pushSync(t, s, e3)
+	require.Equal(t, 0, s.head)
+	require.Equal(t, 0, s.tail)
+	require.Equal(t, 3, s.size())
+
+	// Pushing new elements should drop old data
+	s.dropHandler = func(e *model.ProcessEvent) {
+		model.AssertProcessEvents(t, e1, e)
```

Comment:
[minor] There's nothing that's checking these handlers actually get called. Maybe you should just add these events into a variable and perform an assertions on it outside of the handler, wdyt?